### PR TITLE
test(geo): add geo populate e2e tests

### PR DESCRIPTION
**Pull request by AaronZyLee**
#### Description of changes

Add e2e tests for `amplify geo populate`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    }
+    deleteProjectDir(projRoot);
+  });
+
+  it('should throw error if there is no geofence collection provisioned', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot);
+    await expect(populateGeofencesWithDefault(projRoot)).rejects.toThrowError();
```
Comment (phani-srikar) [inline packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts:34]: Can we move the steps from `initJS...` until `addGeofenceCollection...` to the test setup as we are only testing the populate workflow in all the tests.

File: packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  });
+  it('should upload a valid GeoJSON file with auto-generated root level ID to provisioned geofence collection', async () => {
+    await initJSProjectWithProfile(projRoot, {});
+    await addAuthWithDefault(projRoot);
+    await updateAuthAddUserGroups(projRoot, ['admin']);
+    await addGeofenceCollectionWithDefault(projRoot);
+    await amplifyPushWithoutCodegen(projRoot);
+    await expect(populateGeofencesWithDefault(projRoot, { geoJSONFileName: 'valid-root-level-id.json' })).resolves.not.toThrow();
```
Comment (phani-srikar) [inline packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts:42]: We might need to also verify that the collection now has the geofences in it by using `https://docs.aws.amazon.com/location-geofences/latest/APIReference/API_ListGeofences.html`. You can do this as an enhancement in a different PR if you would like to.

**Review by phani-srikar — CHANGES_REQUESTED** [review]
One change requested and other can be handled in a different PR.

File: packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts (hunk 1 above)
Comment (AaronZyLee) [reply packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts:34]: I am considering compress the similar tests into one case where we provision once and upload multiple geo json files, which will save much time on total run time

Comment (phani-srikar) [reply packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts:34]: That might make debugging difficult in case something fails. I would say as a best practice test one specific case in a test.

Comment (AaronZyLee) [reply packages/amplify-e2e-tests/src/__tests__/geo-populate.test.ts:37]: Update: The cases are still separate but only has one provision run

**Review by phani-srikar — APPROVED** [review]
